cmd/ci-operator/main.go

```diff
@@ -1495,3 +1483,49 @@ func getPullSecretFromFile(filename string) (*coreapi.Secret, error) {
 	secret.Data[coreapi.DockerConfigJsonKey] = src
 	return secret, nil
 }
+
+func getResolverInfo(jobSpec *api.JobSpec, o *options) *load.ResolverInfo {
+	// identify org, repo, and branch from refs
+	var refs prowapi.Refs
+	if jobSpec.Refs != nil {
+		refs = *jobSpec.Refs
+	}
+	info := &load.ResolverInfo{
+		Address: o.resolverAddress,
+		Org:     refs.Org,
+		Repo:    refs.Repo,
+		Branch:  refs.BaseRef,
+		Variant: o.variant,
+	}
+	if len(info.Org) > 0 && len(info.Repo) > 0 && len(info.Branch) > 0 {
+		return info
+	}
+
+	// if org, repo, or branch could not be identified, try extra refs
+	for _, ref := range jobSpec.ExtraRefs {
+		if len(info.Org) == 0 {
+			info.Org = ref.Org
+		}
+		if len(info.Repo) == 0 {
+			info.Repo = ref.Repo
+		}
+		if len(info.Branch) == 0 {
+			info.Branch = ref.BaseRef
+		}
+		if len(info.Org) > 0 && len(info.Repo) > 0 && len(info.Branch) > 0 {
+			return info
+		}
+	}
+
+	// if org, repo, or branch could not be identified in any refs, try flags
```

Comment:
This is the opposite of what I would expect, flags usually override whatever value would be used if they were not present.